### PR TITLE
Add some documentation for podman

### PR DESCRIPTION
**Pull request by abergeron**
This is a draft for the podman docs.

I would welcome comments about missing sections or details that are not explained.

### CODE REVIEW

File: docs/Userguide_containers.rst [hunk 1]
```diff
@@ -3,170 +3,44 @@
...
-
-
-**DO NOT USE IMAGES WITH SENSITIVE INFORMATION** yet, it will soon be possible.
-For now, every image is pulled to a common registry but access-control will soon
-be implemented.
+Generally any command-line argument accepted by docker will work with podman. this means that you can mostly use docker examples in you find on the web by replacing docker with podman in the command line.
 
+.. note::
+    Complete Podman Documentation:
+    https://docs.podman.io/en/stable/
```
Comment (satyaog) [inline docs/Userguide_containers.rst:12]:
```suggestion
.. note::
    Complete Podman Documentation: https://docs.podman.io/en/stable/
```

File: docs/Userguide_containers.rst [hunk 2]
```diff
@@ -3,170 +3,44 @@
...
-
-
-
-Volumes
--------
-
-``/home/yourusername``, ``/Tmp``, ``/ai`` and all ``/network/..`` sub-folders are
-mounted inside the container.
-
+You can use multiple containers in a single job, but you have to be careful about the memory and CPU limits of the job.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:27]:
```suggestion
You can use multiple containers in a single job, but you have to be careful
about the memory and CPU limits of the job.
```

File: docs/Userguide_containers.rst [hunk 3]
```diff
@@ -3,170 +3,44 @@
...
 ---
 
-To access the GPU inside a container, you need to specify ``--module=nvidia`` on
-the ``sbatch/salloc/shifter`` command line
-
-.. prompt:: bash
-
-    shifter --image=centos:7 --module=nvidia bash
-
+To use a GPU you need to a GPU job and then use the ``--device nvidia.com/gpu=all`` for all GPUs allocated to the job or ``--device nvidia.com/gpu=n`` where n is the gpu you want in the container, starting at 0.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:32]:
```suggestion
To use a GPU in a container, you need to a GPU job and then use ``--device
nvidia.com/gpu=all`` to make all GPUs allocated available in the container or
``--device nvidia.com/gpu=N`` where `N` is the gpu index you want in the
container, starting at 0.
```

File: docs/Userguide_containers.rst [hunk 4]
```diff
@@ -3,170 +3,44 @@
 Using containers
 ================
 
-Docker containers are now available on the local cluster with a root-less
-system called Shifter integrated into SLURM.
-*It is still in beta and be careful with this usage*
+Docker containers are now available on the local cluster without root priviledges using `podman <https://podman.io>`_.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:6]:
```suggestion
Docker containers are now available on the local cluster without root
priviledges using `podman <https://podman.io>`_.
```

File: docs/Userguide_containers.rst [hunk 5]
```diff
@@ -3,170 +3,44 @@
...
-
-.. prompt:: bash
-
-    shifterimg images
-
-
-**DO NOT USE IMAGES WITH SENSITIVE INFORMATION** yet, it will soon be possible.
-For now, every image is pulled to a common registry but access-control will soon
-be implemented.
+Generally any command-line argument accepted by docker will work with podman. this means that you can mostly use docker examples in you find on the web by replacing docker with podman in the command line.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:8]:
```suggestion
Generally any command-line argument accepted by docker will work with podman.
This means that you can mostly use the docker examples you find on the web by
replacing `docker` with `podman` in the command line.
```

File: docs/Userguide_containers.rst [hunk 6]
```diff
@@ -3,170 +3,44 @@
...
+.. note::
+    Complete Podman Documentation:
+    https://docs.podman.io/en/stable/
 
 Using in SLURM
 --------------
 
-Containerized Batch job
-^^^^^^^^^^^^^^^^^^^^^^^
+To use podman you can just use the podman command in either a batch script or an interactive job.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:17]:
```suggestion
To use podman you can just use the `podman` command in either a batch script or
an interactive job.
```

File: docs/Userguide_containers.rst [hunk 7]
```diff
@@ -3,170 +3,44 @@
...
 
-You must use the ``--image=docker:image_name:latest`` directive to specify
-the container to use. Once the container is mounted, you are not yet
-inside the container's file-system, you must use the ``shifter`` command
-to execute a command in the chroot environment of the container.
-
-For example:
+One difference in configuration is that for certain technical reasons all the storage for podman (images, containers, ...) is on a job-specific location and will be lost after the job is complete or preempted. If you have data that must be preseved across jobs, you can `mount <https://docs.podman.io/en/v5.2.4/markdown/podman-run.1.html#mount-type-type-type-specific-option>`_ a local folder inside the container, such as $SCRATCH or you home to save data.
```
Comment (satyaog) [inline docs/Userguide_containers.rst:19]:
```suggestion
One difference in configuration is that for certain technical reasons all the
storage for podman (images, containers, ...) is on a job-specific location and
will be lost after the job is complete or preempted. If you have data that must
be preseved across jobs, you can `mount
<https://docs.podman.io/en/v5.2.4/markdown/podman-run.1.html#mount-type-type-type-specific-option>`_
a local folder inside the container, such as `$SCRATCH` or your home to save
data.
```

File: docs/Userguide_containers.rst [hunk 8]
```diff
@@ -3,170 +3,156 @@
 Using containers
 ================
 
-Docker containers are now available on the local cluster with a root-less
-system called Shifter integrated into SLURM.
-*It is still in beta and be careful with this usage*
+Docker containers are now available on the local cluster without root
+priviledges using `podman <https://podman.io>`_.
```
Comment (btravouillon) [inline docs/Userguide_containers.rst:7]:
```suggestion
Podman containers are now available as tech preview on the Mila cluster
without root privileges using `podman <https://podman.io>`_.
```

File: docs/Userguide_containers.rst [hunk 9]
```diff
@@ -3,170 +3,156 @@
...
-    username@eos20:~$ shifter sh
-    ~ $ cat /etc/os-release
-    NAME="Alpine Linux"
-    ID=alpine
-    VERSION_ID=3.10.2
-    PRETTY_NAME="Alpine Linux v3.10"
-
-    ~ $
-
+To use a GPU in a container, you need to a GPU job and then use ``--device
```
Comment (btravouillon) [inline docs/Userguide_containers.rst:53]: You need to "queue" a GPU job ?

**Review by btravouillon — CHANGES_REQUESTED** [review]

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Review by obilaniu — APPROVED** [review]

**Pull request merged**